src/app/services/search.service.ts

```diff
@@ -14,14 +14,19 @@ import { Utils } from 'app/shared/utils/utils';
 export class SearchService {
 
   public isError = false;
-
+  // This might get large
+  private _cachedItems = {};
```

Comment:
This commit will also cache all getItem calls. The only way to cache bust this would be to refresh the page. Might need to think about adding in some cache busting on this object.